src/solvers/FR/Basis.jl

```diff
@@ -333,10 +352,11 @@ function fr_operators(N, sol_pts, cor_fun)
         @assert false
     end
 
-    bl, br = zeros(nd), zeros(nd)
+    T = eltype(xg)
+    bl, br = zeros(T, nd), zeros(T, nd)
     for i in 1:nd
-        bl[i] = 2.0 * dgl(N, 2.0 * xg[i] - 1.0)
-        br[i] = 2.0 * dgr(N, 2.0 * xg[i] - 1.0)
+        bl[i] = T(2) * dgl(N, T(2) * xg[i] - one(T))
```

Comment:
```suggestion
        bl[i] = 2.0f0 * dgl(N, 2.0f0 * xg[i] - 1.0f0)
```

Do it like this when you have hardcoded `Real`. The parametrized types are okay the way you have done it.

Also make sure you did not miss any hardcoded `Real`s.